core/src/main/scala/org/broadinstitute/dsde/rawls/entities/local/LocalEntityProvider.scala

```diff
@@ -65,7 +66,17 @@ class LocalEntityProvider(workspace: Workspace, implicit protected val dataSourc
               logger.info(s"entity statistics cache: miss ($missReason) [${workspaceContext.workspaceIdAsUUID}]")
 
               traceDBIOWithParent("retrieve-uncached-results", outerSpan) { _ =>
-                dataAccess.entityQuery.getEntityTypeMetadata(workspaceContext)
+                dataAccess.entityQuery.getEntityTypeMetadata(workspaceContext) flatMap { metadata =>
+                  val saveCacheAction = if (cacheEnabled && !isEntityCacheCurrent) {
+                    // if the entity cache is not current, AND we have the metadata result, save it to the cache!
+                    // the user has done us the favor of waiting for the result, let's take advantage of that result.
+                    // if saving the cache here fails, ignore the failure and still get the metadata to the user
+                    opportunisticSaveEntityCache(metadata, dataAccess)
+                  } else {
+                    DBIO.successful(())
+                  }
+                  saveCacheAction.map(_ => metadata)
+                }
```

Comment:
I considered moving this to a fire-and-forget operation. I decided against that because it adds code and testing complexity. This extra call will only happen when the cache is out of date, so it shouldn't be all that frequent.